patterns/guaranteed_receiver_nack.go

```diff
@@ -0,0 +1,128 @@
+
+	if err != nil {
+		panic(err)
+	}
+
+	// Connect to the messaging serice
+	if err := messagingService.Connect(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Connected to the broker? ", messagingService.IsConnected())
+
+	if err != nil {
+		panic(err)
+	}
+
+	queueName := "durable-queue"
+	durableExclusiveQueue := resource.QueueDurableExclusive(queueName)
+
+	// Build a Gauranteed message receiver with NACK support and bind to the given queue
+	// Set the required message settlement outcome(s) on the flow using the WithRequiredMessageOutcomeSupport() builder method
+	persistentReceiver, err := messagingService.CreatePersistentMessageReceiverBuilder().
+		WithMessageClientAcknowledgement().
+		// Add message settlement outcomes support on the created Flow here (Failed and Reject for NACK(ing) messages)
+		WithRequiredMessageOutcomeSupport(config.PersistentReceiverFailedOutcome, config.PersistentReceiverRejectedOutcome).
+		Build(durableExclusiveQueue)
+
+	// // Build a Gauranteed message receiver with NACK support and bind to the given queue
+	// // Set the required message settlement outcome(s) on the flow using the configuration provider
+	// persistentReceiver, err := messagingService.CreatePersistentMessageReceiverBuilder().
+	// 	WithMessageClientAcknowledgement().
+	// 	// Add message settlement outcomes support on the created Flow here (Failed and Reject for NACK(ing) messages)
+	// 	FromConfigurationProvider(config.ReceiverPropertyMap{
+	// 		config.ReceiverPropertyPersistentMessageRequiredOutcomeSupport: fmt.Sprintf("%s,%s",config.PersistentReceiverFailedOutcome, config.PersistentReceiverRejectedOutcome),
+	// 	}).
+	// 	Build(durableExclusiveQueue)
+
+	// Handling a panic from a non existing queue
+	defer func() {
+		if err := recover(); err != nil {
+			fmt.Printf("Make sure queue name '%s' exists on the broker.\nThe following error occurred when attempting to connect to create a Persistent Message Receiver:\n%s", queueName, err)
+		}
+	}()
+
+	// Start Persistent Message Receiver
+	if err := persistentReceiver.Start(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Persistent Receiver running? ", persistentReceiver.IsRunning())
+
+	// Message Handler
+	messageHandler := func(message message.InboundMessage) {
+		var messageBody string
+
+		if payload, ok := message.GetPayloadAsString(); ok {
+			messageBody = payload
+		} else if payload, ok := message.GetPayloadAsBytes(); ok {
+			messageBody = string(payload)
+		}
+
+		fmt.Printf("Received Message Body %s \n", messageBody)
+		// fmt.Printf("Message Dump %s \n", message)
+
+		// Settle the message here with one of the three supported settlement outcomes: ACCEPTED, FAILED and REJECTED
+		messageSettlementError := persistentReceiver.Settle(message, config.PersistentReceiverAcceptedOutcome) // Accept(acknowlegde) the message
+		// messageSettlementError := persistentReceiver.Settle(message, config.PersistentReceiverFailedOutcome) // fail the message
```

Comment:
The different settlements should be used in different functions or modules for each type, so that the user actually has a sample of how to use each of the different types

Reply:
Okay. I have made the update